codex/sales/reservations.nim

```diff
@@ -370,54 +375,68 @@ proc createAvailability*(
 
   return success(availability)
 
-proc createReservation*(
+method createReservation*(
   self: Reservations,
   availabilityId: AvailabilityId,
   slotSize: UInt256,
   requestId: RequestId,
   slotIndex: UInt256
-): Future[?!Reservation] {.async.} =
+): Future[?!Reservation] {.async, base.} =
 
-  trace "creating reservation", availabilityId, slotSize, requestId, slotIndex
+  let lock = self.locks.mgetOrPut(availabilityId, newAsyncLock())
```

Comment:
How sure are we that this is the only function where we need to acquire/release a lock for an availabilityId? It seems there are other places in this module that interact with both the availabilities and the reservations.

Reply:
Good point, will recheck.

Reply:
I have fixed those places that touch Availability's sizes. Please have a look.